### PR TITLE
Feature/pcoippm 10

### PR DESCRIPTION
**Pull request by Maheliusz**

### CODE REVIEW

File: server/pools/models.py [hunk 1]
```diff
@@ -1,3 +1,49 @@
...
+    description = models.TextField(default="")
+
+    def already_reserved_slots(self, start, end):
+        """
+        Counts number of slots already taken
+        :param start: Datetime of beginning of reservation
+        :param end: Datetime of finish of reservation
+        :return: Number of slots already reserved
+        """
+        return sum([order.count for order in Reservation.objects.filter(pool_id=self.id)
```
Comment (Lukasz1928) [inline server/pools/models.py]: Rename "order" to "reservation"?

File: server/pools/models.py [hunk 2]
```diff
@@ -1,3 +1,49 @@
...
+    """
+    Model equivalent for one pool of virtual machines
+    """
+    pool_id = models.CharField(max_length=50)
+    displayName = models.CharField(max_length=100)
+    maximumCount = models.PositiveSmallIntegerField(default=0)
+    enabled = models.BooleanField(default=True)
+    description = models.TextField(default="")
+
+    def already_reserved_slots(self, start, end):
```
Comment (Lukasz1928) [inline server/pools/models.py]: Rename to "calculate_already_reserved_slots" or something similar?

File: server/pools/models.py [hunk 3]
```diff
@@ -1,3 +1,49 @@
...
+        return self.maximumCount - self.already_reserved_slots(start, end) >= number_of_slots
+
+
+class Reservation(models.Model):
+    """
+    Stores single reservation; is related to :model:'pools.Pool' and :model:'django.contrib.auth.models.User'
+    """
+    pool = models.ForeignKey(Pool, on_delete=models.CASCADE)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    count = models.PositiveSmallIntegerField(default=0)
```
Comment (Lukasz1928) [inline server/pools/models.py]: Count of what?

File: server/pools/tests.py [hunk 4]
```diff
@@ -1,3 +1,56 @@
...
+from rest_framework.test import APIRequestFactory
+
+from . .models import Pool, Reservation
+
 
 # Create your tests here.
+from . .views import login
+
+
+class SampleTest(TestCase):
```
Comment (Lukasz1928) [inline server/pools/tests.py]: Why?

File: server/pools/views.py [hunk 5]
```diff
@@ -1,4 +1,34 @@
...
+    """
+    if request.method == 'POST':
+        try:
+            json_body = json.loads(request.body)
+            username, password = json_body['username'], json_body['password']
+            assert username is not None and password is not None
+        except (TypeError, AssertionError, KeyError):
+            return HttpResponseBadRequest("Incorrect request body")
+        user = authenticate(username=username, password=password)
+        print(user)
```
Comment (Lukasz1928) [inline server/pools/views.py]: Debug on develop? Or is it supposed to be here for some reason?

File: server/pools/tests.py [hunk 6]
```diff
@@ -1,3 +1,56 @@
...
+        self.user = User.objects.create(username="test", password="test")
+        Reservation.objects.create(pool_id=self.pool.id, user_id=self.user.id, count=5,
+                                   start_datetime=datetime(2018, 4, 1, 11, 15),
+                                   end_datetime=datetime(2018, 4, 1, 12, 45))
+
+    def test_already_reserved_slots(self):
+        self.assertEqual(self.pool.already_reserved_slots(datetime(2018, 4, 1, 11, 00), datetime(2018, 4, 1, 12, 30)),
+                         5)
+
+    def test_can_place_reservation(self):
```
Comment (Lukasz1928) [inline server/pools/tests.py]: Maybe make two separate tests instead of one?

File: server/pcoippoolmanager/settings.py [hunk 7]
```diff
@@ -78,12 +79,13 @@
         'ENGINE': 'django.db.backends.postgresql',
         'NAME': 'postgres',
         'USER': 'postgres',
-        'HOST': 'db',
+        # TODO change host to proper one
```
Comment (Lukasz1928) [inline server/pcoippoolmanager/settings.py]: Who is supposed to do that?

Comment (Maheliusz) [reply server/pcoippoolmanager/settings.py]: DevOps, because I don't know how credentials are supposed to look for the server we'll finally deploy on.

**Pull request merged**